### ISSUE TITLE
5.2.40 breaks database queries on HHVM

### ISSUE THREAD

**Issue by iangcarroll** [laravel/framework#14426]
Upgraded to `5.2.41` and now any database query causes the following error to be thrown on HHVM:

```
SQLSTATE[HY000]: General error: user-supplied class does not have a constructor, use NULL for the ctor_params parameter, or simply omit it (SQL: [...])
```

Looks like `5.2.40` introduced the error: https://laracasts.com/discuss/channels/forge/sqlstatehy000-general-error-user-supplied-class-does-not-have-a-constructor

**Comment by GrahamCampbell** [comment]
What HHVM version please? If the latest, please report this bug to facebook.

**Comment by iangcarroll** [comment]
We're on `3.12-lts`.

**Comment by GrahamCampbell** [comment]
Still facebook's fault for not being php compatible, but re-opening this since we should probably mitigate this ourselves.

**Comment by vlakoff** [comment]
I have submitted a fix for this: #14429.

**Comment by vlakoff** [comment]
HHVM isn't that wrong, as using an empty array is not "documented", in the error message.

Still, it is an incompatibility with PHP.

**Comment by GrahamCampbell** [comment]
HHVM is wrong if it's not exactly the same as PHP. PHP is the definition of being correct.

**Comment by vlakoff** [comment]
Fixed by #14429 and [Laravel 5.2.42](https://github.com/laravel/framework/commit/d1bf46f359d2ede2d4ef847fb79d3758d868f2b7) has been released, so this issue can be closed.
